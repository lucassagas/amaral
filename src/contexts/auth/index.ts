export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
