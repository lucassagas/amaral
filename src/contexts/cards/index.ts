export { CardContext, CardContextProvider } from './cards';

