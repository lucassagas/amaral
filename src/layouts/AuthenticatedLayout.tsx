import { Header } from '@/components/Header'
import { Flex } from '@chakra-ui/react'
import { ReactNode } from 'react'

type AuthenticatedLayoutProps = {
  children: ReactNode
}

export function AuthenticatedLayout({ children }: AuthenticatedLayoutProps) {
  return (
    <Flex direction="column" h="100vh">
      <Header />
      <Flex w="100%" my="6" maxWidth={1480} mx="auto" px="6" overflow="hidden">
        {children}
      </Flex>
    </Flex>
  )
}
