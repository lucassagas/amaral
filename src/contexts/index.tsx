import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
