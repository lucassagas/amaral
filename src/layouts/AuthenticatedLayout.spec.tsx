import { render } from '@testing-library/react'
import { AppProvider } from '../contexts'
import { AuthenticatedLayout } from './AuthenticatedLayout'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: 'example.com',
      query: '',
      asPath: 'example.com',
    }
  },
}))

describe('<AuthenticatedLayout />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <AppProvider>
        <AuthenticatedLayout>
          <h1>Layout</h1>
        </AuthenticatedLayout>
      </AppProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
