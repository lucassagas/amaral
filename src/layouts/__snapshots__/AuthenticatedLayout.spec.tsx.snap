// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticatedLayout /> should render correctly 1`] = `
<div>
  <div
    class="css-s92abg"
  >
    <header
      class="css-5tnzri"
    >
      <button
        aria-label="Open navigation"
        class="chakra-button css-10fhv4q"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 4h18v2H3V4zm0 7h18v2H3v-2zm0 7h18v2H3v-2z"
            />
          </g>
        </svg>
      </button>
      <img
        alt="logo"
        class="chakra-image css-xos48b"
        src="/logo.png"
      />
      <div
        class="css-70qvj9"
      >
        <div
          class="chakra-stack css-zrx1dy"
        >
          <svg
            class="chakra-icon css-1cv0b1u"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M18 8h2a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V9a1 1 0 0 1 1-1h2V7a6 6 0 1 1 12 0v1zM5 10v10h14V10H5zm6 4h2v2h-2v-2zm-4 0h2v2H7v-2zm8 0h2v2h-2v-2zm1-6V7a4 4 0 1 0-8 0v1h8z"
              />
            </g>
          </svg>
          <svg
            class="chakra-icon css-1cv0b1u"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 18h2v2h12V4H6v2H4V3a1 1 0 0 1 1-1h14a1 1 0 0 1 1 1v18a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-3zm2-7h7v2H6v3l-5-4 5-4v3z"
              />
            </g>
          </svg>
        </div>
        <div
          class="css-70qvj9"
        >
          <span
            class="chakra-avatar css-nxs594"
          >
            <svg
              aria-label=" avatar"
              class="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
      </div>
    </header>
    <div
      class="css-zr4pa4"
    >
      <aside
        class="css-1hjbzlu"
      >
        <div
          class="chakra-stack css-1yucvmi"
        >
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1azq1gc"
            >
              Geral
            </p>
            <div
              class="chakra-stack css-1vcm6cv"
            >
              <a
                align="center"
                class="chakra-link css-6qooy7"
                href="/app/dashboard"
              >
                <svg
                  class="chakra-icon css-1cv0b1u"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M13 21V11h8v10h-8zM3 13V3h8v10H3zm6-2V5H5v6h4zM3 21v-6h8v6H3zm2-2h4v-2H5v2zm10 0h4v-6h-4v6zM13 3h8v6h-8V3zm2 2v2h4V5h-4z"
                    />
                  </g>
                </svg>
                <p
                  class="chakra-text css-y7ioq2"
                >
                  Dashboard
                </p>
              </a>
            </div>
          </div>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1azq1gc"
            >
              Admin
            </p>
            <div
              class="chakra-stack css-1vcm6cv"
            >
              <a
                align="center"
                class="chakra-link css-6qooy7"
                href="/users"
              >
                <svg
                  class="chakra-icon css-1cv0b1u"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 7h5v2h-5V7zm-2 5h7v2h-7v-2zm3 5h4v2h-4v-2zM2 22a8 8 0 1 1 16 0h-2a6 6 0 1 0-12 0H2zm8-9c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4z"
                    />
                  </g>
                </svg>
                <p
                  class="chakra-text css-y7ioq2"
                >
                  Usu√°rios
                </p>
              </a>
            </div>
          </div>
        </div>
      </aside>
      <h1>
        Layout
      </h1>
    </div>
  </div>
</div>
`;
